### ISSUE TITLE
Highlight as you type in Find hangs Xed

### ISSUE THREAD

**Issue by mcarans** [linuxmint/xed#207]
```
 * Xed version (xed --version)  1.6.3
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 18.3
```

**Issue**
The highlight as you type feature in Find hangs Xed. When you type the first letter, it attempts to go and highlight that one letter everywhere and then hangs - even on documents that aren't that big on a well specced laptop.

**Steps to reproduce**
Load a document. You got to menu Search and then Find. Type the first character you want to search for. Xed hangs as it tries to highlight that character everywhere.

**Expected behaviour**
I think there needs to be a much longer delay before the highlighting starts after typing a letter as even if I try to type the first and second letter consecutively as quickly as possible, it still gets stuck trying to highlight the first letter everywhere.

Also I think it would be great to have the option to turn off this automatic highlighting as you type.

**Other information**

**Comment by rudolphi** [comment]
This bug makes using the search function unusable even for smaller files, like a 2 MB log file.

**Comment by mcarans** [comment]
The bug still persists in Mint 19.

**Comment by mcarans** [comment]
Still running into this issue with latest Mint.

**Comment by mcarans** [comment]
Still a problem in Mint 19.2.

**Comment by icarter09** [comment]
Linux Mint 19.2 64bit
xed v2.2.3

I'm not seeing any lag or noticing Xed hangs as the first letter is searched and found.

**Comment by mcarans** [comment]
> Linux Mint 19.2 64bit
> xed v2.2.3
> 
> I'm not seeing any lag or noticing Xed hangs as the first letter is searched and found.

How big a file did you use? As another commenter suggested try with a 2Mb log file for example.

**Comment by icarter09** [comment]
When I tested it with a 4Mb file and had no issues.

When I tested it with a 8Mb file, then I experienced the lag. But also experienced Xed crashing because the search was pushing the CPU to 100%

**Comment by mcarans** [comment]
Perhaps you have more powerful hardware than I do? I have an Intel Core i7-8550U laptop.

**Comment by Hordeking** [comment]
I'm experiencing a similar issue on 2.4.2. I don't think it's related to the machine performance, since previous versions seemed to have no trouble with larger files on my laptop. Now I can barely get three characters typed in the find dialog before xed more or less stalls out till I sigkill it.

**Comment by rredford** [comment]
this bug is still here. I usually start with unusual character like $ so I don't accidentally type with 'e' on very large file and crash xed.

**Comment by mcarans** [comment]
More than 4 years have passed since I opened this issue, so I guess it is regarded as low priority.

**Comment by Hordeking** [comment]
> More than 4 years have passed since I opened this issue, so I guess it is regarded as low priority.

I would say so.

**Comment by rredford** [comment]
Generally xed also badly handles large text files. Reminded me of old MS notepad.

**Comment by JosephMcc** [comment]
> Generally xed also badly handles large text files. Reminded me of old MS notepad.

Unfortunately this is true. But it isn't really Xed's fault. The underlying GtkTextView has issues with this and we can't do a lot about that. It's not so much about priority but about what we can actually do. Maybe it would be possible to do something like only search after a certain amount of letters are typed? But I'm not sure what that limit would be and where it would start to be an issue.

**Comment by mcarans** [comment]
@JosephMcc It might be sufficient just to make it so that searching starts after 2 letters since that will hugely reduce the number of matches in most cases.

**Comment by rredford** [comment]
That AND add option to search only after hitting enter. Off by default but yeah

**Comment by rredford** [comment]
> The underlying GtkTextView has issues with this and we can't do a lot about that. It's not so much about priority but about what we can actually do.

Hmm isn't there way to clip data range for display and fake scrollbars (would have to, what with view area actually shorter than what it would be, with entire file). We certainly have room in memory. Just need to have fairly wide range of data clipped from some point above "top of window viewable area" to some point below bottom of window view area.
